zkevm-circuits/src/evm_circuit/execution.rs

```diff
@@ -581,13 +599,21 @@ impl<F: Field> ExecutionConfig<F> {
         stored_expressions_map.insert(G::EXECUTION_STATE, stored_expressions);
 
         // Enforce the logic for this opcode
-        let q_steps: &dyn Fn(&mut VirtualCells<F>) -> Expression<F> =
+        let sel_step: &dyn Fn(&mut VirtualCells<F>) -> Expression<F> =
             &|meta| meta.query_advice(q_step, Rotation::cur());
-        let q_steps_first: &dyn Fn(&mut VirtualCells<F>) -> Expression<F> =
+        let sel_step_first: &dyn Fn(&mut VirtualCells<F>) -> Expression<F> =
             &|meta| meta.query_selector(q_step_first);
+        let sel_step_last: &dyn Fn(&mut VirtualCells<F>) -> Expression<F> =
+            &|meta| meta.query_selector(q_step_last);
+        let sel_not_step_last: &dyn Fn(&mut VirtualCells<F>) -> Expression<F> = &|meta| {
+            meta.query_advice(q_step, Rotation::cur())
+                * (1.expr() - meta.query_selector(q_step_last))
```

Comment:
```suggestion
                * not::expr(meta.query_selector(q_step_last))
```

Reply:
Updated in https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/795/commits/ffbeaf6e168e519926d18d93ae4c0db5a9540161